src/util/AglShell.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 Collabora, Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the
+ * next paragraph) shall be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT.  IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#ifndef __AGL_SHELL_UTIL_H
+#define __AGL_SHELL_UTIL_H
+
+enum agl_shell_surface_type {
```

Comment:
Honestly, I think we should somehow hide these concepts to non-AGL implementation.

I.e. this very specific way of defining the surface parameters is valid for AGL and nothing else. If you move AglShell to the toplevel you will end up manipulating the Agl shell surface parameters in layers that are not AGL specific.

I would prefer an abstract type SurfaceProperties that, in AGL implementation, could define its properties, and use it accordingly.

Reply:
> I would prefer an abstract type SurfaceProperties that, in AGL implementation, could define its properties, and use it accordingly.

Okay, understood.